packages/expo-modules-core/e2e/setup/create-eas-project-macos.ts

```diff
@@ -0,0 +1,363 @@
+  try {
+    // Initialize the macOS project
+    await spawnAsync('npx', ['--yes', 'react-native-macos-init'], {
+      cwd: projectRoot,
+    });
+  } catch {
+    // Ignore Podfile errors on 0.75.2
+  }
+
+  // copy Podfile
+  const podfilePath = path.resolve(dirName, '..', 'fixtures', 'Podfile');
+  await fs.copyFile(podfilePath, path.join(projectRoot, 'macos', 'Podfile'));
+
+  // copy index.js
+  const indexPath = path.resolve(dirName, '..', 'fixtures', 'index.js');
+  await fs.copyFile(indexPath, path.join(projectRoot, 'index.js'));
+
+  // copy metro.config.js
+  const metroConfigPath = path.resolve(dirName, '..', 'fixtures', 'metro.config.js');
+  await fs.copyFile(metroConfigPath, path.join(projectRoot, 'metro.config.js'));
+
+  // copy App.tsx
+  const AppPath = path.resolve(dirName, '..', 'fixtures', 'App.tsx');
+  await fs.copyFile(AppPath, path.join(projectRoot, 'App.tsx'));
+}
+
+/**
+ * Adds all the dependencies and other properties needed for the E2E test app
+ */
+async function preparePackageJson(projectRoot: string, repoRoot: string) {
+  // Create the project subfolder to hold NPM tarballs built from the current state of the repo
+  const dependenciesPath = path.join(projectRoot, 'dependencies');
+  await fs.mkdir(dependenciesPath);
+
+  console.time('Done packing dependencies');
+  for (const dependencyChunk of dependencyChunks) {
+    await Promise.all(
+      dependencyChunk.map(async (dependencyName) => {
+        console.log(`Packing ${dependencyName}...`);
+        console.time(`Packaged ${dependencyName}`);
+        const result = await packExpoDependency(
+          repoRoot,
+          projectRoot,
+          dependenciesPath,
+          dependencyName
+        );
+        expoResolutions[dependencyName] = result.dependency;
+        console.timeEnd(`Packaged ${dependencyName}`);
+      })
+    );
+  }
+  console.timeEnd('Done packing dependencies');
+
+  const extraDevDependencies = {
+    '@config-plugins/detox': '^5.0.1',
+    '@types/express': '^4.17.17',
+    '@types/jest': '^29.4.0',
+    express: '^4.18.2',
+    'form-data': '^4.0.0',
+    jest: '^29.3.1',
+    'jest-circus': '^29.3.1',
+    prettier: '^2.8.1',
+    'ts-jest': '^29.0.5',
+    'patch-package': '^8.0.0',
+  };
+
+  // Remove the default Expo dependencies from create-expo-app
+  let packageJson = JSON.parse(await fs.readFile(path.join(projectRoot, 'package.json'), 'utf-8'));
+  for (const dependencyName of getExpoDependencyNamesForDependencyChunks(dependencyChunks)) {
+    if (packageJson.dependencies[dependencyName]) {
+      delete packageJson.dependencies[dependencyName];
+    }
+  }
+  // Add dependencies and resolutions to package.json
+  packageJson = {
+    ...packageJson,
+    scripts: {
+      postinstall: 'patch-package',
+      ...packageJson.scripts,
+    },
+    dependencies: {
+      ...expoResolutions,
+      ...packageJson.dependencies,
+      'react-native-macos': '0.75.2',
```

Comment:
Can we make it more smart? It should always use the latest patch of the version that we use in the monorepo

Reply:
Sure, we can do that, the only thing is that they don't really use npm tags but we could just get the list of most recent one